### PR TITLE
Disconnect and multi-network

### PR DESCRIPTION
**Pull request by emilbayes**
This PR now encompasses both how disconnects are communicated to dApps and how we handle multiple networks in the wallet.

1. `client.disconnected` is sent to dApps as JSON-RPC notifications when a dApp is removed from the wallet as a "connection". dApps can listen for this with the `window.vega.on('client.disconnected', listener)` API
2. dApps should now call `client.connect_wallet` with a `{ chainId }` param which will prompt the user as part of the approval flow to pick a network configuration that matches the `chainId`. This can be done with the `window.vega.connectWallet({ chainId })` API. For backwards compat, we will pick the default one for the extension if none is given for now and print a warning in DevTools
3. `client.get_chain_id` remains for now, but prints a warning in DevTools that the API is deprecated. This is the `window.vega.getChainId()` method. If the dApp is not yet connected, it will return the chain ID for the default network for the extension, otherwise give the chainId picked during connection.

Changes to the `admin` and interaction APIs:

1. `admin.list_networks` now returns a list of the full network configuration for all networks that are not hidden
2. `popup.review_connection` now also sends the requested `chainId` from a dApp (with fallback for dApps that still don't send one)
3. `popup.review_connection` now expects a reply in the shape of `{ approved: bool; networkId: string }` to store for the connection.

The following migrations will be applied on update:

1. All existing networks stored will be deleted. This was never directly user configurable and we made some changes to the format, which made it easier to just replace the stored networks.
4. We add both Fairground and Mainnet to all builds. The previous "Fairground" build defaults to the fairground testnet, while the "Mainnet" build defaults to mainnet.
5. All existing connections are updated to have the preferred network ID and chain ID from 2. as their prefill. If users wish to change this, they will have to delete the dApp connection and reconnect (after dApps start sending a preferred chain id in the API).

### CODE REVIEW

**Comment by dalebennett1992** [comment]
When connecting to a network that does not exist I get this
image

This is the API call
await vega.connectWallet({chainId: 'vega-fairground-202305051806'})

the correct network is vega-fairground-20230505180**5**

File: specs/1137-CLNT-client_api.md [hunk 1]
```diff
@@ -0,0 +1,45 @@
+# Client API
+
+## Methods
+
+### Connect (client.connect_wallet)
+
+- If I connect and it is not approved then I get an error telling me the user rejected the connection
+- If I connect and it is approved then I do not get an error
+- If I connect and it is approved and I try to connect again then the connection is automatically approved
```
Comment (bwallacee) [inline specs/1137-CLNT-client_api.md:9]: Is there a time limit or anything on this?

With me being new and dumb to browser wallet, this semi reads that if a connection as been approved once, any future connections will never need to be re approved?

All good if that's actually the case and correct.

File: specs/1137-CLNT-client_api.md [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+- If I connect with a non-existent chain Id I get an error telling me the network was not found
+- If I connect with a chainId I am connected to that chain
+- If I connect with a chainId and then connect again with a different chainId then I get an error indicating that I am already connected to a different chain
+
+### Get Chain Id (client.get_chain_id)
+
+- If I am not connected then I receive the selected chainId for the wallet
+- If I connect and then get the chain id then I get the chain id of that network
+
+### Is Connected (client.is_connected)
```
Comment (bwallacee) [inline specs/1137-CLNT-client_api.md:20]: Maybe excessive, but what if a user doesn't approve or reject the connection?

File: specs/1137-CLNT-client_api.md (hunk 1 above)
Comment (dexturr) [reply specs/1137-CLNT-client_api.md:9]: Yep that's correct 😄

File: specs/1137-CLNT-client_api.md (hunk 2 above)
Comment (dexturr) [reply specs/1137-CLNT-client_api.md:20]: Worth adding I shall do so 👍

**Review by daro-maj — APPROVED** [review]

**Review by dexturr — APPROVED** [review]

**Pull request merged**